### ISSUE TITLE
Github "Sign in with browser" open localhost in browser, login fails

### ISSUE THREAD

**Issue by iozcelik** [github/VisualStudio#2521]
## Versions
 - GitHub Extension for Visual Studio version: v2.11.96
 - Visual Studio version: 16.6.0

### Steps to Reproduce
1. Go to 'Team Explorer'
2. Click on 'Connect...' button inside Github panel
3. Click on 'Sign in with browser'
4. Browser open tab with localhost address
5. Connection fails

### Expected behavior

Open to a github.com adress

### Screenshots

![github](https://user-images.githubusercontent.com/10682780/83515676-6d336080-a4de-11ea-82a3-b3df79af5ef4.png)
![adres](https://user-images.githubusercontent.com/10682780/83515736-8a682f00-a4de-11ea-9f80-39835be310be.png)

**Comment by jcansdale** [comment]
Hi @iozcelik,

I'm sorry you were experiencing this! Did you ever manage to get it to work?

**Comment by iozcelik** [comment]
Sign in with browser never worked.

**Comment by techduggu** [comment]
Note: Adding this comment here to provide more information (as already added into #2509 )

Hi Team,

Even I am facing this issue with GitHub extension with Visual Studio 2019. I have tried the above workarounds (uninstalling/reverting to previous version etc.) but no luck.

I am able to sign-in successfully to GitHub on my browser but through extension, when I click on 'Sign in with your browser' option, it opens the browser with the below URL and it just keeps loading:
https://github.com/login/oauth/authorize?client_id=a200baed193bb2088a6e&scope=user%2Crepo%2Cgist%2Cwrite%3Apublic_key%2Cread%3Aorg%2Cworkflow&state=cd3677b8-bf82-4d2d-80c2-71fb3ed7cb63

Below is the error captured from Github Extension log:

> 2020-08-15 22:29:05.832 [20420] INFO [58] GitHubPackage             Initializing GitHub Extension v2.11.104.25275 in Microsoft Visual Studio 2019 (16.6.30114.105)
2020-08-15 22:29:23.860 [20420] EROR [01] LoginTabViewModel         Error logging into '"https://github.com/"' as ''
Octokit.AuthorizationException: Requires authentication
   at Octokit.Connection.HandleErrors(IResponse response)
   at Octokit.Connection.<RunRequest>d__58.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Octokit.Connection.<Run>d__57`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Api.LoginManager.<GetUserAndCheckScopes>d__23.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 346
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Api.LoginManager.<ReadUserWithRetry>d__22.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 335
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Api.LoginManager.<LoginViaOAuth>d__14.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 158
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.VisualStudio.ConnectionManager.<LogInViaOAuth>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.ViewModels.Dialog.LoginTabViewModel.<LoginToHostViaOAuth>d__56.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.App\ViewModels\Dialog\LoginTabViewModel.cs:line 173
{"message":"Requires authentication","documentation_url":"https://docs.github.com/rest/reference/users#get-the-authenticated-user"}
2020-08-15 22:31:05.818 [20420] EROR [05] GitHubServiceProvider     Error loading GitHub.Services.IConnectionManager
System.InvalidOperationException: Due to high risk of deadlock you cannot call GetService from a background thread in an AsyncPackage derived class. You should instead call GetServiceAsync (without calling Result or Wait on the resultant Task object) or switch to the UI thread with the JoinableTaskFactory.SwitchToMainThreadAsync method before calling GetService.
   at Microsoft.VisualStudio.Shell.AsyncPackage.GetService(Type serviceType)
   at Microsoft.VisualStudio.Shell.Package.System.IServiceProvider.GetService(Type serviceType)
   at GitHub.VisualStudio.GitHubServiceProvider.TryGetService(Type serviceType)
2020-08-15 22:34:32.192 [20420] EROR [01] LoginTabViewModel         Error logging into '"https://github.com/"' as ''
Octokit.AuthorizationException: Requires authentication
   at Octokit.Connection.HandleErrors(IResponse response)
   at Octokit.Connection.<RunRequest>d__58.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Octokit.Connection.<Run>d__57`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Api.LoginManager.<GetUserAndCheckScopes>d__23.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 346
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Api.LoginManager.<ReadUserWithRetry>d__22.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 335
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.Api.LoginManager.<LoginViaOAuth>d__14.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 158
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.VisualStudio.ConnectionManager.<LogInViaOAuth>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at GitHub.ViewModels.Dialog.LoginTabViewModel.<LoginToHostViaOAuth>d__56.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.App\ViewModels\Dialog\LoginTabViewModel.cs:line 173
{"message":"Requires authentication","documentation_url":"https://docs.github.com/rest/reference/users#get-the-authenticated-user"}

**Comment by jcansdale** [comment]
@iozcelik @techduggu,

Are you by any chance using a VPN?

**Comment by iozcelik** [comment]
@jcansdale I tried and result is same.

**Comment by jcansdale** [comment]
@iozcelik,

> I tried and result is same.

You tried with and without a VPN?

**Comment by iozcelik** [comment]
When I open this issue, I tried without Vpn. After you asked, I tried with it.

**Comment by jcansdale** [comment]
@iozcelik,

> When I open this issue, I tried without Vpn. After you asked, I tried with it.

That's a shame. 😢 

When the page fails to load:

![image](https://user-images.githubusercontent.com/11719160/94801653-43337180-03de-11eb-9f2d-b50cf15ced24.png)

Could you try replacing `localhost` with `127.0.0.1`? Could you also try `ping localhost` and check that's it's pointing at `127.0.0.1`?

Thanks!

**Comment by jcansdale** [comment]
Hi @iozcelik,

Another user was able to fix this by launching Visual Studio as Administrator before logging in, see https://github.com/github/VisualStudio/issues/2550#issuecomment-703280956.

Could you give that a try? 🙏

**Comment by iozcelik** [comment]
@jcansdale I will try as soon as possible.

**Comment by avatar25** [comment]
It doesn't work for me as well. I tried to run VS19 as an administrator but it doesn't resolve. With/without VPN makes no difference either.

**Comment by jcansdale** [comment]
@avatar25 thanks for checking!

Do you know if you've used any apps that run on localhost successfully?

A quick check would to try using my TestDriven.NET extension:
https://testdriven.net/download_release.aspx?LicenceType=Personal

After installing the extension, you can right-click inside any method and `Run Test(s)`. If this works, we'll have confirmed that a callback to localhost can work on your machine.

**Comment by dotnetgik** [comment]
I resolved issue by copying the link

 https://github.com/login/oauth/authorize?client_id=a200baed193bb2088a6e&scope=user%2Crepo%2Cgist%2Cwrite%3Apublic_key%2Cread%3Aorg%2Cworkflow&state=710e9410-38be-4566-a024-36fe91e414ad

 and opening it in Incognito Mode now i am able to login and load my repositories properly in the visual studio

**Comment by jcansdale** [comment]
@dotnetgik interesting, thanks! I wonder why Incognito Mode was required? 🤔

**Comment by KyuuzoT** [comment]
I've faced the same problem with authorization as everyone above. So, I've tried to:
1. Run VS2019 as administrator.
2. Clean the .git-credentials
3. Set my wireless connection as "private"
4. Restart computer several times.
5. Reinstall GitHub Extension several times.
6. Reinstall entire VS2019 several times.
7. Copy connection link to Chrome with incognito mode.
8. Copy connection link to Firefox and Edge.

Also, I haven't seen if somebody mentioned before but I see in TeamExplorer my local repositories (not all of them, just from one specific folder - C:\Users\User\Source\Repos). However I can not look through or work with any of my remote projects.

So, what should I do to make this extension work?

**Comment by KyuuzoT** [comment]
@KyuuzoT I've just tried to connect via VPN - no result.
Then, I've tried to copy authorization link to TOR Browser (yeah, I know it's Firefox) - A window for entering an email and password appeared, after entering my credentials I've received a verification code. But after entering the code I've got the same "localhost refused to connect"-problem.

**Comment by SenzoNgele** [comment]
Lool after I struggled for years.. I have found a Solution to This
The course is. First I use my google chrome which I saved my Github account with.
So my VS was taking me to Microsoft Edge and get the button disabled.

All you have to do is to change you Default browser on your machine to the one u have signed is with on github

**Comment by jcansdale** [comment]
Hi @KyuuzoT,

I'm sorry you've had such a hard time getting this to work!

There was another suggestion here https://github.com/github/VisualStudio/issues/2527#issuecomment-730082783.

1. Put this into Chrome address bar: `chrome://net-internals/#hsts`
2. Then go to the bottom and delete `localhost` from the Delete domain security policies

Might that the the issue?

**Comment by KyuuzoT** [comment]
> Hi @KyuuzoT,
> 
> I'm sorry you've had such a hard time getting this to work!
> 
> There was another suggestion here [#2527 (comment)](https://github.com/github/VisualStudio/issues/2527#issuecomment-730082783).
> 
> 1. Put this into Chrome address bar: `chrome://net-internals/#hsts`
> 2. Then go to the bottom and delete `localhost` from the Delete domain security policies
> 
> Might that the the issue?

Yeah, I forgot to mention that I've already done this and still have no result.

Also, I've been testing extension in it's current state on my computer for some time and found out that I could make commits, create branches, pull & fetch etc if I'll find project that fits my remote repo and open it from VS2019 directly. Also I can clone my remote repos via direct link instead of opening them, which is not conveniently, because I do not need several versions of one project in different folders on my local PC (I'm using version management system for this particular reason, right?). So I have functionality of this extension despite the fact that I still got the "unconnected" message and I still can not see any of my remote repos in VS2019 (I see only list of my local repos). What does it mean and how it works this way I have no idea at all.

**Comment by iozcelik** [comment]
After update to lastest version the problem is fix for me. Now I can login.

**Comment by KyuuzoT** [comment]
@iozcelik "the latest version" of what? VS or Git extension?

**Comment by iozcelik** [comment]
@KyuuzoT  "Visual Studio 16.8.2" but now I have lots of different problem :)

**Comment by KyuuzoT** [comment]
@iozcelik Well, I'm facing this problem while being on VS 16.8.2

**Comment by jcansdale** [comment]
Hi @KyuuzoT,

I've just added a new subcommand `set-token` to a .NET Core tool called `ghvs`. This tool contains various GitHub and Visual Studio related subcommands. You can find the PR here:
https://github.com/jcansdale/ghvs/pull/44

Please can you try the following:

1. Create a PAT with the [following](https://github.com/github/VisualStudio/blob/824bab4ab2c3d8f6787cf38ff5ff9d4e9df00e98/src/GitHub.Api/ApiClientConfiguration.cs#L43) scopes: "user", "repo", "gist", "write:public_key", "read:org", "workflow"
You can use this link https://github.com/settings/tokens
![image](https://user-images.githubusercontent.com/11719160/100374454-37da7c00-3004-11eb-8182-20d9245ea115.png)
2. Install the `ghvs` tool
`dotnet tool install -g ghvs --version 1.0.113-g022796defc`
3. Set the GitHub for Visual Studio token
`ghvs set-token --secret-store ghfvs <PAT>`
4. Open Visual Studio and `File > Open > Open from GitHub...`

If everything went to plan, a list of repositories should appear. 🤞 

Please let me know how you get on!

**Comment by KyuuzoT** [comment]
Hi @jcansdale 

I've got this message:
`Unrecognized command or argument 'set-token'

Did you mean this?
    access-token
`
on step №3. And I don't see neither a command nor the option 'set-token' in ghvs --help.

**Comment by jcansdale** [comment]
@KyuuzoT,

Could you check that you installed version `1.0.113-g022796defc` (which is the version from jcansdale/ghvs#44)?

```
$ dotnet tool install -g ghvs --version 1.0.113-g022796defc
You can invoke the tool using the following command: ghvs
Tool 'ghvs' (version '1.0.113-g022796defc') was successfully installed.
```

The help page should look like this:

```
$ ghvs --help

Usage: ghvs [options] [command]

Options:
  --help          Show help information
  --host          The host URL
  --access-token  The access token to use
  --secret-store  The secret store to use (Git or GHfVS)

Commands:
  branch          Show information about the current branch
  install         Install 'x-github-client' protocol handler
  issues          Show issues
  login           Login using GitHub Credential Manager
  logout          Logout using GitHub Credential Manager
  open            Open a file or folder in Visual Studio
  open-url        Open a GitHub URL in Visual Studio
  orgs            Show visible organizations (requires 'read:org' scope)
  pulls           Show pull requests
  repos           List repositories
  set-token       Set credentials with a personal access token
  uninstall       Uninstall 'x-github-client' protocol handler
  upstream        Show information about the upstream repository
  viewer          Show viewer information

Run 'ghvs [command] --help' for more information about a command.
```

**Comment by V4A001** [comment]
@jcansdale for me the fix with ghvs works. Thank you! With azure az login I have same issue, I use az login --use-device-code instead.

**Comment by chrisdyckgeorgian** [comment]
I am having the same challenge. The GitHub link shows "This site can't be reached"

**Comment by jcansdale** [comment]
Hi @chrisdyckgeorgian,

This dialog is actually part of recent Visual Studio versions and isn't installed with GitHub for Visual Studio.

Could you try using the `File > Open > Open from GitHub ...` command and see if you can login/clone a repository? (this functionality is part of GitHub for Visual Studio)

**Comment by chrisdyckgeorgian** [comment]
@jcansdale the same problem occurs. It does bring up the GitHub sign in but when I go to the Sign In with browser option, I get the same error.

![image](https://user-images.githubusercontent.com/47536011/101812017-d4167f80-3ae8-11eb-9275-0d38fd53797a.png)

**Comment by jsnape** [comment]
I think this works but is it supposed to show the sign-in with browser prompt every time you go a git action that connects to the remote? Its very annoying.

**Comment by chrisdyckgeorgian** [comment]
@jsnape Yes, very annoying! Helpfully the next round of VS will fix this.

**Comment by gintarevitke** [comment]
Hi,
I have the same problem now after GitHub client was updated to 2.29.2.2. Visual Studio version 16.8.3 and GitHub extention for Visual Studio 2.11.106.19330.

**Comment by OmidRafiee** [comment]
If after logging you will be redirected to localhost instead of github :

![123](https://user-images.githubusercontent.com/25098596/105668009-b9119a80-5ef1-11eb-8f46-a57d1333f451.png)



I solved it this way :

1- Put this into Chrome address bar:  chrome://net-internals/#hsts
2- Then go to the bottom and delete **localhost** from the Delete domain security policies

![Untitled](https://user-images.githubusercontent.com/25098596/105668024-c0d13f00-5ef1-11eb-8a14-2be727b21809.png)

**Comment by jsnape** [comment]
So 

> I think this works but is it supposed to show the sign-in with browser prompt every time you go a git action that connects to the remote? Its very annoying.

So this week I'm working across 22 repos. So every time I want to do anything its 22 login dialogs to click:

![image](https://user-images.githubusercontent.com/1582936/105870176-ecd2ea00-5fef-11eb-9dda-88a8e41c9dff.png)

Which very quickly turns into: 

![image](https://user-images.githubusercontent.com/1582936/105870329-125ff380-5ff0-11eb-81fa-11627d67f12e.png)

So right now this is completely unusable for me.

**Comment by eduardoalba0** [comment]
> So
> 
> > I think this works but is it supposed to show the sign-in with browser prompt every time you go a git action that connects to the remote? Its very annoying.
> 
> So this week I'm working across 22 repos. So every time I want to do anything its 22 login dialogs to click:
> 
> ![image](https://user-images.githubusercontent.com/1582936/105870176-ecd2ea00-5fef-11eb-9dda-88a8e41c9dff.png)
> 
> Which very quickly turns into:
> 
> ![image](https://user-images.githubusercontent.com/1582936/105870329-125ff380-5ff0-11eb-81fa-11627d67f12e.png)
> 
> So right now this is completely unusable for me.

Same here

**Comment by jcansdale** [comment]
@eduardoalba0, @jsnape, @OmidRafiee, @chrisdyckgeorgian, @gintarevitke, hi folks 👋🏻 

I've been in touch with the Visual Studio team who are maintaining the GitHub login functionality. They've asked if you could report this issue via Visual Studio's built in `Help > Send Feedback > Report a Problem...` functionality.

You can read about it here:
https://devblogs.microsoft.com/visualstudio/overhauling-the-visual-studio-feedback-system/

The feedback form includes an easy way to submit a screen recording, which I'm sure would be very useful!

Obviously the ability to login is super important and it looks like it isn't working for a group of users. I've tried, but haven't been able to reproduce the issue.

If you could take the time to report it, that would be very much appreciated! Please let me know when you submit it, so I can give my contacts the heads up.

Thanks!

**Comment by prasadsonawane-MB** [comment]
facing same issue

**Comment by peanut0001** [comment]
fucking dogshit extension dogshit company go fix your dogshit software

**Comment by Lucas7yoshi** [comment]
Unable to sign in using any listed information here.

**Comment by edika99** [comment]
Same here. Totally unusable extension

**Comment by chrisdyckgeorgian** [comment]
Yes, totally useless. Please fix! This huge change should have had much more user input before going to production...

**Comment by ttaylor29** [comment]
> Hi @KyuuzoT,
> 
> I'm sorry you've had such a hard time getting this to work!
> 
> There was another suggestion here [#2527 (comment)](https://github.com/github/VisualStudio/issues/2527#issuecomment-730082783).
> 
> 1. Put this into Chrome address bar: `chrome://net-internals/#hsts`
> 2. Then go to the bottom and delete `localhost` from the Delete domain security policies
> 
> Might that the the issue?

@jcansdale This situation fixed it for me. I think my work computer has some policies that cause this. Thanks!

**Comment by vodemn** [comment]
Having same issue!

**Comment by jcansdale** [comment]
Hi folks,

I'm sorry this issue has been such a problem! If you're running into this issue, please could you do the following:

1. Uninstall `GitHub for Visual Studio`.
From Developer PowerShall for VS execute:
`vsixinstaller /uninstall:c3d3dc68-c977-411f-b3e8-03b0dccf7dfc`
2. Open Visual Studio and `Help > Send Feedback > Report a Problem...`
3. Record feedback for the following command `File > Clone Repository... > GitHub`

Connecting to GitHub with the built in functionality should look like this:
![image](https://user-images.githubusercontent.com/11719160/110671672-b7144a00-81c6-11eb-9db3-6a85f4670b6d.png)

Please let me know if you run into any issues doing this!

//cc @vodemn @chrisdyckgeorgian @edika99 @Lucas7yoshi @peanut0001 @prasadsonawane-MB @yossiy123

**Comment by Lucas7yoshi** [comment]
Unfortunately no luck with that either. 

Same localhost issue 
![image](https://user-images.githubusercontent.com/28872095/110703750-23407f00-81c2-11eb-9c56-a76c99a303b9.png)
@jcansdale

**Comment by jcansdale** [comment]
Hi @Lucas7yoshi 👋🏻 ,

> Unfortunately no luck with that either.

This is part of the GitHub login that is now built into Visual Studio (rather than being part of the extension). Could you use the `Help > Send Feedback > Report a Problem...` feature to report it?

If a few people report this issue, we'll have a much better chance of getting it fixed! 🤞🏻

**Comment by Lucas7yoshi** [comment]
I'll report it shortly when I have a chance.

**Comment by rmatejka** [comment]
I'm having the same problem in VS 2019 Preview Version 16.10.0 Preview 1.0. I've tried all of the suggestions and still no go. The logs contain this:

2021-03-15 17:47:01.507 [04260] INFO [67] GitHubPackage             Initializing GitHub Extension v2.11.102.28613 in Microsoft Visual Studio 2019 Preview (16.10.31025.218)
2021-03-15 18:23:49.749 [13012] INFO [14] GitHubPackage             Initializing GitHub Extension v2.11.102.28613 in Microsoft Visual Studio 2019 Preview (16.10.31025.218)
2021-03-15 18:24:02.155 [13012] EROR [21] GitHubServiceProvider     Error loading GitHub.Services.IConnectionManager
System.InvalidOperationException: Due to high risk of deadlock you cannot call GetService from a background thread in an AsyncPackage derived class. You should instead call GetServiceAsync (without calling Result or Wait on the resultant Task object) or switch to the UI thread with the JoinableTaskFactory.SwitchToMainThreadAsync method before calling GetService.
   at Microsoft.VisualStudio.Shell.AsyncPackage.GetService(Type serviceType)
   at Microsoft.VisualStudio.Shell.Package.System.IServiceProvider.GetService(Type serviceType)
   at GitHub.VisualStudio.GitHubServiceProvider.TryGetService(Type serviceType)


Any suggestions?

**Comment by jcansdale** [comment]
@rmatejka I'm interested to know what happens if you uninstall the extension and try Visual Studio's built-in GitHub integration. See https://github.com/github/VisualStudio/issues/2521#issuecomment-795792333 for instructions.

Please let me know how you get on!

**Comment by legistek** [comment]
This has been opened since June 2020 and still isn't fixed? What a joke.

**Comment by rmatejka** [comment]
> @rmatejka I'm interested to know what happens if you uninstall the extension and try Visual Studio's built-in GitHub integration. See [#2521 (comment)](https://github.com/github/VisualStudio/issues/2521#issuecomment-795792333) for instructions.
> 
> Please let me know how you get on!

The same issue just different port:
http://localhost:47120/?code=16a457fd323681d0dc61&state=fea7bdc119a44d77be40f99c4523b9e6

**Comment by legistek** [comment]
I finally determined what was causing this for me and how to fix it.

I frequently run my application server on localhost. My application is hard coded to send the `Strict-Transport-Security` header even in debug mode (which it probably need not be). This causes Chrome to refuse to access the domain (i.e. localhost) except through SSL. It remembers this for the future, even when it's a different application on localhost. Thus the github integration hook, which apparently only uses HTTP, gets rejected by Chrome.

Easy solution - clear your browser history and cache (you don't have to clear cookies). And that's it.

Correct solution for Microsoft - run the hook through HTTPS.

**Comment by rajoo8** [comment]
> I finally determined what was causing this for me and how to fix it.
> 
> I frequently run my application server on localhost. My application is hard coded to send the `Strict-Transport-Security` header even in debug mode (which it probably need not be). This causes Chrome to refuse to access the domain (i.e. localhost) except through SSL. It remembers this for the future, even when it's a different application on localhost. Thus the github integration hook, which apparently only uses HTTP, gets rejected by Chrome.
> 
> Easy solution - clear your browser history and cache (you don't have to clear cookies). And that's it.
> 
> Correct solution for Microsoft - run the hook through HTTPS.

Thanks a ton, this works for me finally!

**Comment by clinton-foote** [comment]
Works in Edge also.

edge://net-internals/#hsts

**Comment by mightysabean** [comment]
What worked for me was deleting all my cookies and browsing history (From Microsoft Edge Browser)

**Comment by AnkitSaraf30553** [comment]
> Note: Adding this comment here to provide more information (as already added into #2509 )
> 
> Hi Team,
> 
> Even I am facing this issue with GitHub extension with Visual Studio 2019. I have tried the above workarounds (uninstalling/reverting to previous version etc.) but no luck.
> 
> I am able to sign-in successfully to GitHub on my browser but through extension, when I click on 'Sign in with your browser' option, it opens the browser with the below URL and it just keeps loading: https://github.com/login/oauth/authorize?client_id=a200baed193bb2088a6e&scope=user%2Crepo%2Cgist%2Cwrite%3Apublic_key%2Cread%3Aorg%2Cworkflow&state=cd3677b8-bf82-4d2d-80c2-71fb3ed7cb63
> 
> Below is the error captured from Github Extension log:
> 
> > 2020-08-15 22:29:05.832 [20420] INFO [58] GitHubPackage             Initializing GitHub Extension v2.11.104.25275 in Microsoft Visual Studio 2019 (16.6.30114.105)
> > 2020-08-15 22:29:23.860 [20420] EROR [01] LoginTabViewModel         Error logging into '"https://github.com/"' as ''
> > Octokit.AuthorizationException: Requires authentication
> > at Octokit.Connection.HandleErrors(IResponse response)
> > at Octokit.Connection.d__58.MoveNext()
> > --- End of stack trace from previous location where exception was thrown ---
> > at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
> > at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
> > at Octokit.Connection.d__57`1.MoveNext() --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    at GitHub.Api.LoginManager.<GetUserAndCheckScopes>d__23.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 346 --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    at GitHub.Api.LoginManager.<ReadUserWithRetry>d__22.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 335 --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    at GitHub.Api.LoginManager.<LoginViaOAuth>d__14.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 158 --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    at GitHub.VisualStudio.ConnectionManager.<LogInViaOAuth>d__14.MoveNext() --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    at GitHub.ViewModels.Dialog.LoginTabViewModel.<LoginToHostViaOAuth>d__56.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.App\ViewModels\Dialog\LoginTabViewModel.cs:line 173 {"message":"Requires authentication","documentation_url":"https://docs.github.com/rest/reference/users#get-the-authenticated-user"} 2020-08-15 22:31:05.818 [20420] EROR [05] GitHubServiceProvider     Error loading GitHub.Services.IConnectionManager System.InvalidOperationException: Due to high risk of deadlock you cannot call GetService from a background thread in an AsyncPackage derived class. You should instead call GetServiceAsync (without calling Result or Wait on the resultant Task object) or switch to the UI thread with the JoinableTaskFactory.SwitchToMainThreadAsync method before calling GetService.    at Microsoft.VisualStudio.Shell.AsyncPackage.GetService(Type serviceType)    at Microsoft.VisualStudio.Shell.Package.System.IServiceProvider.GetService(Type serviceType)    at GitHub.VisualStudio.GitHubServiceProvider.TryGetService(Type serviceType) 2020-08-15 22:34:32.192 [20420] EROR [01] LoginTabViewModel         Error logging into '"https://github.com/"' as '' Octokit.AuthorizationException: Requires authentication    at Octokit.Connection.HandleErrors(IResponse response)    at Octokit.Connection.<RunRequest>d__58.MoveNext() --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)    at Octokit.Connection.<Run>d__57`1.MoveNext()
> > --- End of stack trace from previous location where exception was thrown ---
> > at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
> > at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
> > at GitHub.Api.LoginManager.d__23.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 346
> > --- End of stack trace from previous location where exception was thrown ---
> > at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
> > at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
> > at GitHub.Api.LoginManager.d__22.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 335
> > --- End of stack trace from previous location where exception was thrown ---
> > at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
> > at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
> > at GitHub.Api.LoginManager.d__14.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.Api\LoginManager.cs:line 158
> > --- End of stack trace from previous location where exception was thrown ---
> > at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
> > at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
> > at GitHub.VisualStudio.ConnectionManager.d__14.MoveNext()
> > --- End of stack trace from previous location where exception was thrown ---
> > at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
> > at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
> > at GitHub.ViewModels.Dialog.LoginTabViewModel.d__56.MoveNext() in D:\a\VisualStudio\VisualStudio\src\GitHub.App\ViewModels\Dialog\LoginTabViewModel.cs:line 173
> > {"message":"Requires authentication","documentation_url":"[https://docs.github.com/rest/reference/users#get-the-authenticated-user"}](https://docs.github.com/rest/reference/users#get-the-authenticated-user%22%7D)

You can try 3 things for this issue
1. try with Administrator mode of visual studio.
2. download latest git credentials manager
3. add /***/.vs/***/DesignTimeBuild/.dtbcache.v2 to git ignore

**Comment by AnkitSaraf30553** [comment]
You can try 3 things for this issue

try with Administrator mode of visual studio.
download latest git credentials manager
add //.vs//DesignTimeBuild/.dtbcache.v2 to git ignore

**Comment by noesisterna** [comment]
> Lool después de haber luchado durante años ... He encontrado una solución a esto El curso es. Primero uso mi Google Chrome con el que guardé mi cuenta de Github. Entonces, mi VS me estaba llevando a Microsoft Edge y desactivaba el botón.
> 
> Todo lo que tiene que hacer es cambiar su navegador predeterminado en su máquina al que ha firmado en github

Gracias!!! soluciones mi problema!!

**Comment by gnguyen83** [comment]
> I finally determined what was causing this for me and how to fix it.
> 
> I frequently run my application server on localhost. My application is hard coded to send the `Strict-Transport-Security` header even in debug mode (which it probably need not be). This causes Chrome to refuse to access the domain (i.e. localhost) except through SSL. It remembers this for the future, even when it's a different application on localhost. Thus the github integration hook, which apparently only uses HTTP, gets rejected by Chrome.
> 
> Easy solution - clear your browser history and cache (you don't have to clear cookies). And that's it.
> 
> Correct solution for Microsoft - run the hook through HTTPS.

Thank you very much. I got the issue finally resolved with your suggestion. Well done!

**Comment by mlhein** [comment]
> > I finally determined what was causing this for me and how to fix it.
> > I frequently run my application server on localhost. My application is hard coded to send the `Strict-Transport-Security` header even in debug mode (which it probably need not be). This causes Chrome to refuse to access the domain (i.e. localhost) except through SSL. It remembers this for the future, even when it's a different application on localhost. Thus the github integration hook, which apparently only uses HTTP, gets rejected by Chrome.
> > Easy solution - clear your browser history and cache (you don't have to clear cookies). And that's it.
> > Correct solution for Microsoft - run the hook through HTTPS.
> 
> Thank you very much. I got the issue finally resolved with your suggestion. Well done!

Thank you, this resolved the issue.

**Comment by deanparkr** [comment]
I have two machines and neither can sign into GitHub, same issue as this thread, it opens the browser (Chrome) and I sign into GitHub but then get::


_This site can’t be reached localhost took too long to respond.

ERR_TIMED_OUT_

Using latest VS2022. Tried incognito mode, clearing cache, checking HSTS doesnt have localhost listed, tried setting another default browser. No dice.

One of the machines is a new machine with everything installed fresh and has same issue right out of the box :(

**Comment by nehasapra21** [comment]
> Hi @KyuuzoT,
> 
> I'm sorry you've had such a hard time getting this to work!
> 
> There was another suggestion here [#2527 (comment)](https://github.com/github/VisualStudio/issues/2527#issuecomment-730082783).
> 
> 1. Put this into Chrome address bar: `chrome://net-internals/#hsts`
> 2. Then go to the bottom and delete `localhost` from the Delete domain security policies
> 
> Might that the the issue?

Thanks this solution resolve my problem

**Comment by AbstractionsAs** [comment]
None of the proposed solutions seem to solve the problem for me.
It hangs attempting to reach localhost - it doesn't seem to to listening for incoming requests.

**Comment by TechGeneralista** [comment]
I never had a problem logging in, now I have. It was so much that I had to delete the chrome browsing history. solved

**Comment by Premier-Solutions** [comment]
The issue that I was facing was that github was redirecting me to localhost without adding the port number.
http://localhost/?browser_session_id=96cee3401f9a78b6f307eb0819606d4f80e21988a8a945fb6d3b9b74e032de8e&code=90255ccef1a34eac35cb&state=189a50d1-b1ae-405d-8b46-ecf54e3528a1

I just added the port number as mentioned in ttaylor29 (42549) comment and it worked fine then.

**Comment by Alex8695** [comment]
> The issue that I was facing was that github was redirecting me to localhost without adding the port number. http://localhost/?browser_session_id=96cee3401f9a78b6f307eb0819606d4f80e21988a8a945fb6d3b9b74e032de8e&code=90255ccef1a34eac35cb&state=189a50d1-b1ae-405d-8b46-ecf54e3528a1
> 
> I just added the port number as mentioned in ttaylor29 (42549) comment and it worked fine then.

I can say that this fixed my issue as well.

changed localhost to localhost:42549 and it connected without an further issue.

**Comment by asmak9** [comment]
> > The issue that I was facing was that github was redirecting me to localhost without adding the port number. http://localhost/?browser_session_id=96cee3401f9a78b6f307eb0819606d4f80e21988a8a945fb6d3b9b74e032de8e&code=90255ccef1a34eac35cb&state=189a50d1-b1ae-405d-8b46-ecf54e3528a1
> > I just added the port number as mentioned in ttaylor29 (42549) comment and it worked fine then.
> 
> I can say that this fixed my issue as well.
> 
> changed localhost to localhost:42549 and it connected without an further issue.

This fix my issue, but, how can we map the port to automatically open in GitHub sign-In with browser for visual studio?

**Comment by DevMatSolutions** [comment]
> > The issue that I was facing was that github was redirecting me to localhost without adding the port number. http://localhost/?browser_session_id=96cee3401f9a78b6f307eb0819606d4f80e21988a8a945fb6d3b9b74e032de8e&code=90255ccef1a34eac35cb&state=189a50d1-b1ae-405d-8b46-ecf54e3528a1
> > I just added the port number as mentioned in ttaylor29 (42549) comment and it worked fine then.
> 
> I can say that this fixed my issue as well.
> 
> changed localhost to localhost:42549 and it connected without an further issue.

worked for me also

**Comment by santoshasc** [comment]
Fixed 42549 my issue THnks

**Comment by 3PR** [comment]
I found a workaround that worked for me.  (update: it just looked like it authorized but did not)

from GitHub website repository : Open with visual studio.

A popup on visual studio should appear to login.

Select the option to sign in with code.

Enter the code into the webpage that opens and authorize Visual Studio.

Never mind it just looked like it worked. the above comment solution worked (adding port 42549 to localhost)

**Comment by LePoeTe** [comment]
> The issue that I was facing was that github was redirecting me to localhost without adding the port number. http://localhost/?browser_session_id=96cee3401f9a78b6f307eb0819606d4f80e21988a8a945fb6d3b9b74e032de8e&code=90255ccef1a34eac35cb&state=189a50d1-b1ae-405d-8b46-ecf54e3528a1
> 
> I just added the port number as mentioned in ttaylor29 (42549) comment and it worked fine then.

This is fixed my issue.
Thanks

**Comment by camaro322hp** [comment]
> The issue that I was facing was that github was redirecting me to localhost without adding the port number. http://localhost/?browser_session_id=96cee3401f9a78b6f307eb0819606d4f80e21988a8a945fb6d3b9b74e032de8e&code=90255ccef1a34eac35cb&state=189a50d1-b1ae-405d-8b46-ecf54e3528a1
> 
> I just added the port number as mentioned in ttaylor29 (42549) comment and it worked fine then.

Thank you - worked for me also.

**Comment by wk3d** [comment]
This happened to me as well. I tried different things from this and other threads and nothing worked. **Firefox was the only browser taking me to GitHub login so I changed my default browser to Firefox so it would open from VS**. It worked and logged me in. Maybe helpful to try the link in different browsers to find which one actually opens the site.

**Comment by GusGusGusGus** [comment]
In my case I had an issue with hsts as edge and chrome were trying ot use https to redirect to localhost. I just went to edge://net-internals/#hsts within edge and under Delete domain security policies I typed localhost. Then I retried logging in with VS and the authentication was successful.

**Comment by remustechco** [comment]
> In my case I had an issue with hsts as edge and chrome were trying ot use https to redirect to localhost. I just went to edge://net-internals/#hsts within edge and under Delete domain security policies I typed localhost. Then I retried logging in with VS and the authentication was successful.

This is the solution that works for me too.

**Comment by sweenig** [comment]
> https://github.com/github/VisualStudio/issues/2521#issuecomment-701336332

This is what worked for me. Got off my VPN and it authenticated automatically (I was already signed in through my browser).
